lasagne/layers/base.py

```diff
@@ -300,3 +307,50 @@ def get_output_for(self, inputs, **kwargs):
         `NotImplementedError`.
         """
         raise NotImplementedError
+
+
+class UnLayer(MergeLayer):
+
+    """
+    lasagne.layers.UnLayer(incoming, layer, **kwargs)
+
+    The :class:`UnLayer` class performing inverse operations
+    for a single layer of a neural network: transposed layer
+    for a :class:`DenseLayer`, deconvolutional layer for
+    :class:`Conv2DLayer`, :class:`Conv1DLayer`; or
+    an unpooling layer for :class:`MaxPool2DLayer`.
```

Comment:
"performing" -> "performs"

Also please add a sentence on *how* it does this. Something like: "It does so by applying the partial derivative of the layer to be inverted with respect to its input."